addons/account/models/account_invoice.py

```diff
@@ -560,8 +562,21 @@ def _prepare_tax_line_vals(self, line, tax):
             'account_analytic_id': tax['analytic'] and line.account_analytic_id.id or False,
             'account_id': self.type in ('out_invoice', 'in_invoice') and (tax['account_id'] or line.account_id.id) or (tax['refund_account_id'] or line.account_id.id),
         }
+        # If the taxes generate moves on the same financial account as the invoice line,
+        # propagate the analytic account from the invoice line to the tax line.
+        # This is necessary in situations were (part of) the taxes cannot be reclaimed,
+        # to ensure the tax move is allocated to the proper analytic account.
+        if not val.['account_analytic_id'] and line.account_analytic_id and val['account_id'] == line.account_id.id:
+            val['account_analytic_id'] = line.account_analytic_id.id
         return vals
 
+    def _cleaned_tax_grouped(self, tax_grouped):
```

Comment:
Ok, the argument her is valid: it is a whole class of hooks that need to manipulate the tax totals, base totals, etc. This is not something common in europe, but in almost all SA countries we have "withholding logic", most of them only apply after reaching a certain minimum amount per invoice. This makes the ground for an even more general minimum approach.